### PR TITLE
feat: link to JC modal for JC projects

### PR DESCRIPTION
**Pull request by johnnyd-eth**
- Implements an `isJuicecrowd` hook and uses in 6 prior relevant cases
- Implements a modal which opens automatically when loading a Juicecrowd project

DEMO (note: "JuiceCrowd" casing has been fixed to show "Juicecrowd") 
Screen Shot 2023-11-29 at 9 18 45 pm

Screen Shot 2023-11-29 at 9 18 35 pm

### CODE REVIEW

File: src/components/DomainBadge.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ export type DomainBadgeProps = {
   projectId?: number
 }
 
-function getJuicecrowdUrl(projectId: number) {
+export function getJuicecrowdUrl(projectId: number) {
```
Comment (wraeth-eth) [inline src/components/DomainBadge.tsx:15]: This should probably be moved somewhere better where it makes sense

**Review by wraeth-eth — CHANGES_REQUESTED** [review]

File: src/components/DomainBadge.tsx (hunk 1 above)
Comment (johnnyd-eth) [reply src/components/DomainBadge.tsx]: Good call sir, have moved to a `utils/juicecrowd.ts` file

File: src/components/v2v3/V2V3Project/ProjectDashboard/components/JuiceCrowdRedirectModal.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+export function JuicecrowdRedirectModal() {
+  const { projectId } = useProjectMetadata()
+
+  const isJuicecrowd = useIsJuicecrowd()
+  const [open, setOpen] = useState<boolean>(isJuicecrowd)
+
+  if (!projectId) return null
+
+  return (
+    <Modal
```
Comment (tomquirk) [inline src/components/v2v3/V2V3Project/ProjectDashboard/components/JuiceCrowdRedirectModal.tsx:19]: **suggestion (non-blocking)** we do have JuiceModal we can/should use instead.

Maybe a follow-up?

File: src/components/v2v3/V2V3Project/ProjectDashboard/components/JuiceCrowdRedirectModal.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      open={open}
+      onCancel={() => setOpen(false)}
+      width={600}
+      okButtonProps={{ hidden: true }}
```
Comment (tomquirk) [inline src/components/v2v3/V2V3Project/ProjectDashboard/components/JuiceCrowdRedirectModal.tsx:23]: @johnnyd-eth we can pass in `href` to this and I think it'll Just Work - no need for custom buttons

File: src/components/v2v3/V2V3Project/ProjectDashboard/components/JuiceCrowdRedirectModal.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        <Button
+          type="default"
+          onClick={() => setOpen(false)}
+          className="w-full grow"
+        >
+          <Trans>Cancel</Trans>
+        </Button>
+        <Button type="primary" className="w-full grow">
+          <ExternalLink href={getJuicecrowdUrl(projectId)} className="">
+            <Trans>Open on Juicecrowd</Trans>
```
Comment (tomquirk) [inline src/components/v2v3/V2V3Project/ProjectDashboard/components/JuiceCrowdRedirectModal.tsx:55]:
```suggestion
            <Trans>Go to Juicecrowd</Trans>
```

File: src/components/v2v3/V2V3Project/ProjectDashboard/components/JuiceCrowdRedirectModal.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        <div className="mb-3 flex h-20 w-20 items-center justify-center rounded-full bg-bluebs-50 dark:bg-bluebs-900">
+          <Image
+            src="/assets/images/juicecrowd-logo.webp"
+            alt="Juicecrowd logo"
+            height={35}
+            width={35}
+          />
+        </div>
+        <div className="text-xl font-medium">
+          <Trans>It looks like this is a Juicecrowd project.</Trans>
```
Comment (tomquirk) [inline src/components/v2v3/V2V3Project/ProjectDashboard/components/JuiceCrowdRedirectModal.tsx:36]:
```suggestion
          <Trans>View on Juicecrowd</Trans>
```

wdyt?

File: src/components/v2v3/V2V3Project/ProjectDashboard/components/JuiceCrowdRedirectModal.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        </div>
+        <p className="text-base">
+          <Trans>
+            Some functionality may not work on juicebox.money, check it out on
+            Juicecrowd for the full experience.
```
Comment (tomquirk) [inline src/components/v2v3/V2V3Project/ProjectDashboard/components/JuiceCrowdRedirectModal.tsx:41]:
```suggestion
            Some functionality may be missing here. Check out this project on Juicecrowd.gg.
```

**Review by tomquirk — APPROVED** [review]
@johnnyd-eth imma just approve and merge, left some comments if u care to address but dont fuss

**Pull request merged**